### PR TITLE
Initial checking of Rust crate for contracts and sample contract

### PR DESCRIPTION
**Pull request by mattgeddes**
This is just an initial pass. The actual structs and interfaces are expected to change somewhat, but it demonstrates the simplicity of a contract (examples/*.rs) and works as a crate.

### CODE REVIEW

File: examples/contract-split-evenly.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+use anyhow::Result;
+use versatus_rust::versatus_rust::{ComputeInputs, ComputeOutputs, ComputeTransaction};
+
+fn main() -> Result<()> {
+    // Gather the smart contract inputs from stdin
+    let input = ComputeInputs::gather()?;
+
+    // Do contract stuff to generate proposed transactions
+    let mut transactions: Vec<ComputeTransaction> = vec![];
```
Comment (eureka-cpu) [inline examples/contract-split-evenly.rs:9]: just a style nit, if you're creating a new empty vec it's cleaner to use 
```rust
let mut transactions: Vec<ComputeTransaction> = Vec::new();
// or
let mut transactions = Vec::<ComputeTransaction>::new();
```
I usually only use the macro in instances where I don't necessarily need to initialize a new variable. Like in a function call that takes a vec that I've yet to initialize but I will never actually get back from the function in the current scope:
```rust
fn innie<T>(v: Vec<T>) -> NewType<Vec<T>> { ... }

fn outie() {
    let new_type = innie(vec!["head"]);
}
```

File: examples/contract-split-evenly.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let amount_each: u64 =
+        input.application_input.amount / input.application_input.recipients.len() as u64;
+    for recipient in input.application_input.recipients.iter() {
+        let txn = ComputeTransaction {
+            recipient: recipient.to_string(),
+            amount: amount_each,
+        };
+        transactions.push(txn);
```
Comment (eureka-cpu) [inline examples/contract-split-evenly.rs:17]: small style nit
no need to store in a variable here, just push the `ComputeTransaction` directly
```rust
    for recipient in input.application_input.recipients.iter() {
        transactions.push(ComputeTransaction::from_recipient(
            &recipient,
            amount_each,
        ));
    }
```

**Review by eureka-cpu — APPROVED** [review]
Looks good, happy to approve. 

Left a few style suggestions, and as always please run `cargo clippy` to avoid warning build up. 🙂

**Comment by mattgeddes** [comment]
Running `cargo clippy` gave no warnings.

`matthew@prickle:~/src/versatus/versatus-rust$ cargo clippy
   Compiling proc-macro2 v1.0.66
   Compiling unicode-ident v1.0.11
   Compiling serde v1.0.188
   Compiling anyhow v1.0.75
   Compiling serde_json v1.0.105
    Checking ryu v1.0.15
    Checking itoa v1.0.9
   Compiling quote v1.0.33
   Compiling syn v2.0.31
   Compiling serde_derive v1.0.188
    Checking versatus-rust v0.1.0 (/home/matthew/src/versatus/versatus-rust)
    Finished dev [unoptimized + debuginfo] target(s) in 3.49s
`

**Pull request merged**